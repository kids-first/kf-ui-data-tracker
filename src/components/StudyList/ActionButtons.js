import React, {useState} from 'react';
import {Link} from 'react-router-dom';
import {useQuery, useMutation} from '@apollo/react-hooks';
import {ADD_COLLABORATOR, CREATE_REFERRAL_TOKEN} from '../../state/mutations';
import {GET_STUDY_BY_ID, MY_PROFILE} from '../../state/queries';
import {Amplitude} from '@amplitude/react-amplitude';
import {Button, Label, Icon, Popup, Table} from 'semantic-ui-react';
import {
  countStudyNotification,
  countProjectNotification,
} from '../../common/notificationUtils';
import {hasPermission} from '../../common/permissions';
import CavaticaLogo from '../../assets/CavaticaLogo';
import AddCollaboratorModal from '../../modals/AddCollaboratorModal';

/**
 * A button that displays a popup when hovered and optionally a notification
 * label in the right corner.
 */
const PopupButton = ({
  header,
  content,
  icon,
  label = null,
  hoverable = false,
  disabled = false,
  ...props
}) => (
  <Amplitude
    eventProperties={inheritedProps => ({
      ...inheritedProps,
      scope: inheritedProps.scope
        ? [...inheritedProps.scope, 'button', header.toLowerCase()]
        : ['button', header.toLowerCase()],
    })}
  >
    {({logEvent}) => (
      <Popup
        disabled={disabled}
        header={header}
        content={content}
        hoverable={hoverable}
        position="top right"
        trigger={
          <Button
            icon={icon}
            style={{position: 'relative'}}
            content={label}
            onClick={() => logEvent('click')}
            {...props}
          />
        }
      />
    )}
  </Amplitude>
);
/**
 * Contains a button group for study actions
 */
const ActionButtons = ({study}) => {
  const [showCollaborators, setShowCollaborators] = useState(false);

  // Get current user's profile to determine permission level
  const {data: profileData, loading} = useQuery(MY_PROFILE);
  const user = profileData && profileData.myProfile;

  const studyInfoNotif = countStudyNotification(study);
  const projectNotif = countProjectNotification(study);

  const [addCollaborator] = useMutation(ADD_COLLABORATOR, {
    refetchQueries: [
      {
        query: GET_STUDY_BY_ID,
        variables: {
          kfId: study.kfId,
        },
      },
    ],
  });
  const [createToken] = useMutation(CREATE_REFERRAL_TOKEN);

  if (loading) return <span>loading</span>;

  return (
    <Amplitude
      eventProperties={inheritedProps => ({
        ...inheritedProps,
        scope: inheritedProps.scope
          ? [...inheritedProps.scope, 'study actions']
          : ['study actions'],
      })}
    >
      <Table.Cell singleLine width="1">
        <Button.Group icon basic>
          <PopupButton
            header="Study Information"
            icon="info"
            as={Link}
            to={`/study/${study.kfId}/basic-info/info`}
            content={
              hasPermission(user, 'change_study') &&
              studyInfoNotif > 0 && (
                <p>
                  <Icon name="warning sign" /> Missing {studyInfoNotif} fields
                </p>
              )
            }
            label={
              hasPermission(user, 'change_study') &&
              studyInfoNotif > 0 && (
                <Label empty corner circular color="orange" />
              )
            }
          />
          <PopupButton
            header="Documents"
            icon="file"
            as={Link}
            to={`/study/${study.kfId}/documents`}
          />
          <PopupButton
            header="Cavatica Projects"
            icon={<CavaticaLogo className="mr-5 vertical-middle" />}
            as={Link}
            to={`/study/${study.kfId}/cavatica`}
            content={
              hasPermission(user, 'link_project') &&
              projectNotif > 0 && (
                <p>
                  <Icon name="warning sign" /> Missing {projectNotif} required
                  projects
                </p>
              )
            }
            label={
              hasPermission(user, 'link_project') &&
              projectNotif > 0 && <Label empty corner circular color="orange" />
            }
          />
          <PopupButton
            hoverable
            header="Collaborators"
            icon="users"
            content={
              hasPermission(user, 'add_collaborator') && (
                <>
                  <p>Manage collaborators in this study</p>
                  <Button
                    primary
                    fluid
                    content="Add Collaborator"
                    icon="add"
                    labelPosition="left"
                    onClick={() => setShowCollaborators(true)}
                  />
                </>
              )
            }
            as={Link}
            to={`/study/${study.kfId}/collaborators`}
          />
        </Button.Group>
      </Table.Cell>
      <AddCollaboratorModal
        study={study}
        open={showCollaborators}
        onCloseDialog={() => setShowCollaborators(false)}
        addCollaborator={addCollaborator}
        inviteCollaborator={createToken}
      />
    </Amplitude>
  );
};

export default ActionButtons;
