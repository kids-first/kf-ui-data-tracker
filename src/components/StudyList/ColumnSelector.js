import React from 'react';
import {Checkbox, Icon, Item, Popup} from 'semantic-ui-react';

const ColumnSelector = ({selected, available, onChange}) => {
  const change = col => {
    if (selected.includes(col)) {
      selected.splice(selected.indexOf(col), 1);
      onChange([...selected]);
    } else {
      onChange([...selected, col]);
    }
  };

  return (
    <Popup
      basic
      position="bottom"
      on="click"
      trigger={
        <span className="ui dropdown">
          Change Columns
          <Icon name="dropdown" />
        </span>
      }
      header="Select Columns to Display"
      content={Object.keys(available).map(col => (
        <Item key={col}>
          <Checkbox
            label={available[col]}
            checked={selected.includes(col)}
            onChange={() => change(col)}
          >
            {col}
          </Checkbox>
        </Item>
      ))}
    />
  );
};

export default ColumnSelector;
