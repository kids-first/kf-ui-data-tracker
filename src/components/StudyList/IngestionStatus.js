import React from 'react';
import {useQuery, useMutation} from '@apollo/react-hooks';
import {Table} from 'semantic-ui-react';
import {MY_PROFILE, GET_STUDY_BY_ID} from '../../state/queries';
import {CHANGE_INGESTION_STATUS} from '../../state/mutations';
import {hasPermission} from '../../common/permissions';
import StatusSelector from './StatusSelector';

const IngestionStatus = ({study}) => {
  const {data: profileData, loading: profileLoading} = useQuery(MY_PROFILE);
  const [changeStatus, {loading: changing}] = useMutation(
    CHANGE_INGESTION_STATUS,
    {
      refetchQueries: [
        {
          query: GET_STUDY_BY_ID,
          variables: {
            kfId: study.kfId,
          },
        },
      ],
    },
  );

  const user = profileData && profileData.myProfile;

  const onChange = (ev, {value}) =>
    changeStatus({variables: {study: study.id, data: {status: value}}});

  const options = [
    {
      key: 'UNKNOWN',
      text: 'Unknown',
      value: 'UNKNOWN',
    },
    {
      key: 'NOTSTART',
      text: 'Not Started',
      value: 'NOTSTART',
    },
    {
      key: 'INPROG',
      text: 'In Progress',
      value: 'INPROG',
    },
    {
      key: 'COMPLETE',
      text: 'Complete',
      value: 'COMPLETE',
    },
  ];

  if (!profileLoading && hasPermission(user, 'change_ingestion_status')) {
    return (
      <Table.Cell
        singleLine
        width="1"
        positive={['COMPLETE'].includes(study.ingestionStatus)}
        error={['UNKNOWN'].includes(study.ingestionStatus)}
        warning={['INPROG'].includes(study.ingestionStatus)}
        data-cy="ingestion status cell"
      >
        <StatusSelector
          loading={changing}
          value={study.ingestionStatus}
          options={options}
          onChange={onChange}
        />
      </Table.Cell>
    );
  }

  return (
    <Table.Cell
      singleLine
      width="1"
      positive={['COMPLETE'].includes(study.ingestionStatus)}
      error={['UNKNOWN'].includes(study.ingestionStatus)}
      warning={['INPROG'].includes(study.ingestionStatus)}
      data-cy="ingestion status cell"
    >
      {options.find(op => op.key === study.ingestionStatus).text}
    </Table.Cell>
  );
};

export default IngestionStatus;
