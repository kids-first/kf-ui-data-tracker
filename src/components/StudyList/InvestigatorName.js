import React from 'react';
import {Popup, Table} from 'semantic-ui-react';

const InvestigatorName = ({investigatorName}) => {
  // Return an empty cell
  if (!investigatorName)
    return (
      <Table.Cell singleLine width="1">
        -
      </Table.Cell>
    );
  // Return the full investigator's name
  if (investigatorName.length <= 20)
    return (
      <Table.Cell singleLine width="1">
        {investigatorName}
      </Table.Cell>
    );

  // The name is too long so truncate and put full content in a popup
  const name = investigatorName.substring(0, 20);
  return (
    <Popup
      inverted
      hoverable
      position="top center"
      trigger={
        <Table.Cell
          singleLine
          width="1"
          onClick={e => {
            e.preventDefault();
            e.stopPropagation();
          }}
        >
          {name}...
        </Table.Cell>
      }
      content={investigatorName}
    />
  );
};

export default InvestigatorName;
