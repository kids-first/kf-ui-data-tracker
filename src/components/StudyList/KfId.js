import React, {useState} from 'react';
import {Amplitude} from '@amplitude/react-amplitude';
import {Icon, Popup, Table} from 'semantic-ui-react';
import {CopyToClipboard} from 'react-copy-to-clipboard';

const KfId = ({kfId}) => {
  const [copied, setCopied] = useState(false);

  return (
    <Amplitude
      eventProperties={inheritedProps => ({
        ...inheritedProps,
        scope: inheritedProps.scope
          ? [...inheritedProps.scope, 'kfid']
          : ['kfid'],
      })}
    >
      {({logEvent}) => (
        <Popup
          inverted
          position="top left"
          trigger={
            <CopyToClipboard
              text={kfId}
              onCopy={() => {
                setCopied(true);
                logEvent('copy');
                setTimeout(() => {
                  setCopied(false);
                }, 700);
              }}
            >
              <Table.Cell singleLine width="1">
                <code>{kfId}</code>
              </Table.Cell>
            </CopyToClipboard>
          }
          content={
            copied ? <Icon name="check" color="green" /> : 'Copy to clipboard'
          }
        />
      )}
    </Amplitude>
  );
};

export default KfId;
