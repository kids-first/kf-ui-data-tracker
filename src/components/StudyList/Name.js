import React from 'react';
import {Link} from 'react-router-dom';
import {Amplitude} from '@amplitude/react-amplitude';
import {Header, Label, Table} from 'semantic-ui-react';
import defaultAvatar from '../../assets/defaultAvatar.png';

const StudyName = ({study}) => {
  // TODO: Filter out only users in the Investigators group
  const investigators =
    study.collaborators.edges.length &&
    study.collaborators.edges.map(({node}) => node);

  return (
    <Amplitude
      eventProperties={inheritedProps => ({
        ...inheritedProps,
        scope: inheritedProps.scope
          ? [...inheritedProps.scope, 'study name']
          : ['study name'],
      })}
    >
      {({logEvent}) => (
        <Table.Cell
          selectable
          className="overflow-cell-container"
          textAlign="left"
        >
          <Link
            to={'/study/' + study.kfId + '/basic-info/info'}
            onClick={() => logEvent('click')}
            className="overflow-cell"
          >
            <Header size="medium" alt={study.name}>
              {study.name}
              <Header.Subheader>
                <Investigators investigators={investigators} />
              </Header.Subheader>
            </Header>
          </Link>
        </Table.Cell>
      )}
    </Amplitude>
  );
};

const Investigators = ({investigators}) => {
  if (investigators.length) {
    return (
      <Label.Group>
        {investigators.map(user => (
          <Label image key={user.id}>
            <img alt={user.username} src={user.picture || defaultAvatar} />
            {user.username}
          </Label>
        ))}
      </Label.Group>
    );
  }
  return <span>No investigators</span>;
};

export default StudyName;
