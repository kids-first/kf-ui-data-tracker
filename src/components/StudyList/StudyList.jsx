import React, {useState} from 'react';
import PropTypes from 'prop-types';
import classes from 'classnames';
import {withRouter} from 'react-router-dom';
import StudyCard from './StudyCard';
import {Icon, GridContainer} from 'kf-uikit';
import StudyGrid from './StudyGrid';
import ToggleButtons from '../ToggleButtons/ToggleButtons';

/**
 * Displays unordered studies in grid view (include empty stage message)
 */
const StudyList = ({className, studyList, loading}) => {
  const [view, setView] = useState('grid');
  const title = loading ? 'Loading studies ...' : 'Browse Studies';

  return (
    <div className="bg-lightGrey min-h-screen py-32">
      <GridContainer className={studyListClass}>
        <h1 className="text-blue font-title row-1 cell-12">
          {title}
          {!loading ? (
            <ToggleButtons
              onToggle={({text}) => {
                setView(text.toLowerCase());
              }}
              buttons={[
                {text: 'Grid', icon: 'resources'},
                {text: 'list', icon: 'study'},
              ]}
            />
          ) : null}
        </h1>
        {view === 'grid' ? (
          <StudyGrid {...{loading, studyList}} />
        ) : (
          <h2>list</h2>
        )}
      </GridContainer>
    </div>
  );
};

StudyList.propTypes = {
  /** Any additional classes to be applied to the study list*/
  className: PropTypes.string,
  /** Array of study object*/
  studyList: PropTypes.array,
  /** Loading state of the studyList*/
  loading: PropTypes.bool,
};

StudyList.defaultProps = {
  className: null,
  studyList: [],
};

export default StudyList;
