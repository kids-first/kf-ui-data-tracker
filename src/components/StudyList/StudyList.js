import React, {useState} from 'react';
import PropTypes from 'prop-types';
import {Amplitude} from '@amplitude/react-amplitude';
import StudyTable from './StudyTable';
import {Link} from 'react-router-dom';
import {
  Container,
  Segment,
  Grid,
  Header,
  Placeholder,
  Input,
  Button,
  Checkbox,
  Popup,
  Responsive,
} from 'semantic-ui-react';
import {hasPermission} from '../../common/permissions';
import ColumnSelector from './ColumnSelector';

/**
 * A skeleton placeholder for the loading state of the study list header
 */
const HeaderSkeleton = () => (
  <Container as={Segment} basic>
    <Placeholder>
      <Placeholder.Header>
        <Placeholder.Line />
      </Placeholder.Header>
    </Placeholder>
  </Container>
);

/**
 * Displays unordered studies in grid view (include empty stage message)
 */
const StudyList = ({studyList, loading, activeView, history, myProfile}) => {
  const [responsiveClass, setResponsiveClass] = useState('');
  const [searchString, setSearchString] = useState('');
  const [myStudies, setMyStudies] = useState(
    localStorage.getItem('onlyMyStudies') !== null
      ? JSON.parse(localStorage.getItem('onlyMyStudies'))
      : true,
  );
  const [fullWidth, setFullWidth] = useState(
    localStorage.getItem('fullWidth') !== null
      ? JSON.parse(localStorage.getItem('fullWidth'))
      : false,
  );

  const toggleMyStudies = logEvent => {
    setMyStudies(!myStudies);
    logEvent('toggle ' + (myStudies ? 'off' : 'on'));
    localStorage.setItem('onlyMyStudies', !myStudies);
  };

  const toggleWidth = logEvent => {
    setFullWidth(!fullWidth);
    logEvent('toggle ' + (fullWidth ? 'off' : 'on'));
    localStorage.setItem('fullWidth', !fullWidth);
  };

  // Try to restore the column state from local storage or fall back to the
  // defaults if non are found
  // We track the version off the column state so that we may override it in
  // the future if the schema ever changes
  const existingState = JSON.parse(localStorage.getItem('studyColumns'));
  const defaultState = {
    version: 4,
    columns: [
      {key: 'kfId', name: 'Kids First ID', visible: true},
      {key: 'externalId', name: 'phsid/External ID', visible: false},
      {
        key: 'anticipatedSamples',
        name: 'Expected Samples',
        visible: false,
      },
      {
        key: 'sequencingStatus',
        name: 'Sequencing Status',
        visible: false,
      },
      {
        key: 'ingestionStatus',
        name: 'Ingestion Status',
        visible: false,
      },
      {
        key: 'phenotypeStatus',
        name: 'Phenotype Status',
        visible: false,
      },
      {key: 'version', name: 'Version', visible: true},
      {key: 'slackChannel', name: 'Slack Channel', visible: false},
      {key: 'bucket', name: 'Bucket', visible: false},
    ],
    sorting: {
      column: 'name',
      direction: 'descending',
    },
  };
  const [columns, setColumns] = useState(
    existingState && existingState.version === defaultState.version
      ? existingState
      : defaultState,
  );

  const handleSort = column => () => {
    const direction =
      columns.sorting.column !== column
        ? 'ascending'
        : columns.sorting.direction === 'ascending'
        ? 'descending'
        : 'ascending';
    setColumns({
      ...columns,
      sorting: {column, direction},
    });
    localStorage.setItem(
      'studyColumns',
      JSON.stringify({
        columns: existingState ? existingState.columns : defaultState.columns,
        version: defaultState.version,
        sorting: {column, direction},
      }),
    );
  };

  if (loading && !studyList.length) {
    return (
      <Container as={Segment} basic>
        <HeaderSkeleton />
        <Placeholder>
          <Placeholder.Header image>
            <Placeholder.Line />
            <Placeholder.Line />
          </Placeholder.Header>
          <Placeholder.Paragraph>
            <Placeholder.Line />
            <Placeholder.Line />
            <Placeholder.Line />
            <Placeholder.Line />
          </Placeholder.Paragraph>
        </Placeholder>
      </Container>
    );
  }
  const myStudyList =
    myProfile && myProfile.studies.edges.length > 0
      ? myProfile.studies.edges.map(({node}) => node.kfId)
      : [];

  const conactCollaborators = collaborators => {
    return collaborators.length > 0
      ? collaborators
          .map(({node}) => [node.username, node.displayName].join(' '))
          .join(' ')
      : '';
  };

  const filteredStudyList = () => {
    const originList = myStudies
      ? studyList.filter(({node}) => myStudyList.includes(node.kfId))
      : studyList;
    const filteredList = originList.filter(
      ({
        node: {
          name,
          shortName,
          kfId,
          collaborators: {edges},
        },
      }) =>
        [name, shortName, kfId, conactCollaborators(edges)]
          .join(' ')
          .toLowerCase()
          .includes(searchString.toLowerCase()),
    );
    return filteredList;
  };

  return (
    <>
      <Grid as={Segment} basic container>
        <Grid.Row columns={6}>
          <Grid.Column computer={3} tablet={5} mobile={16}>
            <Header as="h1" floated="left">
              Your Studies
            </Header>
          </Grid.Column>
          <Grid.Column
            className="pl-0"
            computer={3}
            tablet={3}
            mobile={6}
            verticalAlign="middle"
            textAlign="right"
          >
            <ColumnSelector
              columns={columns.columns}
              onChange={cols => {
                const newCols = {...columns, columns: cols};
                localStorage.setItem('studyColumns', JSON.stringify(newCols));
                setColumns(newCols);
              }}
            />
          </Grid.Column>
          <Grid.Column
            className="pl-0"
            computer={3}
            tablet={4}
            mobile={5}
            verticalAlign="middle"
            textAlign="right"
          >
            {myProfile && hasPermission(myProfile, 'view_study') && (
              <Amplitude
                eventProperties={inheritedProps => ({
                  ...inheritedProps,
                  scope: inheritedProps.scope
                    ? [
                        ...inheritedProps.scope,
                        'toggle button',
                        'only my studies',
                      ]
                    : ['toggle button', 'only my studies'],
                })}
              >
                {({logEvent}) => (
                  <Checkbox
                    label="Show only my studies"
                    checked={myStudies}
                    onClick={() => toggleMyStudies(logEvent)}
                    data-cy="toggle my studies"
                  />
                )}
              </Amplitude>
            )}
          </Grid.Column>
          <Grid.Column
            computer={3}
            tablet={4}
            mobile={5}
            verticalAlign="middle"
          >
            {myProfile && hasPermission(myProfile, 'add_study') && (
              <Button
                basic
                primary
                fluid
                icon="add"
                content="Add Study"
                as={Link}
                to={`/study/new-study/info`}
              />
            )}
          </Grid.Column>
          <Grid.Column
            className="pl-0"
            tablet={15}
            computer={3}
            mobile={15}
            verticalAlign="middle"
          >
            <Responsive
              as={Input}
              columns={1}
              fireOnMount
              onUpdate={(e, {width}) =>
                setResponsiveClass(
                  width >= Responsive.onlyComputer.minWidth ? '' : 'mt-15',
                )
              }
              className={responsiveClass}
              fluid
              aria-label="search studies"
              iconPosition="left"
              icon="search"
              placeholder="Search by study name or collaborator"
              onChange={(e, {value}) => {
                setSearchString(value);
              }}
              value={searchString}
            />
          </Grid.Column>
          <Grid.Column
            className="pl-0"
            tablet={1}
            computer={1}
            mobile={1}
            verticalAlign="middle"
            textAlign="right"
          >
            <Amplitude
              eventProperties={inheritedProps => ({
                ...inheritedProps,
                scope: inheritedProps.scope
                  ? [...inheritedProps.scope, 'toggle button', 'full width']
                  : ['toggle button', 'full width'],
              })}
            >
              {({logEvent}) => (
                <Popup
                  inverted
                  content="Toggle full width view"
                  position="top right"
                  trigger={
                    <Button
                      className={responsiveClass}
                      data-cy="toggle width button"
                      active={fullWidth}
                      onClick={() => toggleWidth(logEvent)}
                      icon={fullWidth ? 'compress' : 'expand'}
                    />
                  }
                />
              )}
            </Amplitude>
          </Grid.Column>
        </Grid.Row>
      </Grid>
      <Grid as={Segment} container={!fullWidth} basic>
        <Grid.Row>
          {filteredStudyList().length > 0 ? (
            <Grid.Column>
              <StudyTable
                myProfile={myProfile}
                loading={loading}
                studyList={filteredStudyList()}
                columns={columns}
                handleSort={handleSort}
              />
            </Grid.Column>
          ) : (
            <Grid.Column>
              <Header as="h4" disabled textAlign="center">
                No Studies matching your search term. Try searching by Study
                Name
              </Header>
            </Grid.Column>
          )}
        </Grid.Row>
      </Grid>
    </>
  );
};

StudyList.propTypes = {
  /** Array of study object*/
  studyList: PropTypes.array,
  /** Loading state of the studyList*/
  loading: PropTypes.bool,
  /** view to show (grid | list) */
  activeView: PropTypes.oneOf(['grid', 'list']),
};

StudyList.defaultProps = {
  studyList: [],
  activeView: 'list',
};

export default StudyList;
