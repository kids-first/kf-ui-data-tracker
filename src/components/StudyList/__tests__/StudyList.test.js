import React from 'react';
import {MemoryRouter} from 'react-router-dom';
import {render} from 'react-testing-library';
import allStudies from './allStudies.json';
import StudyList from '../StudyList';

it('renders correctly', () => {
  const studies = allStudies.data.allStudies.edges;

  const tree = render(
    <MemoryRouter>
      <StudyList studyList={studies} />
    </MemoryRouter>,
  );
  expect(tree.container).toMatchSnapshot();

  const cards = tree.container.querySelectorAll('.StudyList li');
  expect(cards.length).toBe(3);
});

it('renders loading state', () => {
  const tree = render(
    <MemoryRouter>
      <StudyList loading={true} />
    </MemoryRouter>,
  );
  expect(tree.container).toMatchSnapshot();

  const cards = tree.container.querySelectorAll('.StudyList li');
  // Should contain 4 cards in loading state
  expect(cards.length).toBe(4);
});
