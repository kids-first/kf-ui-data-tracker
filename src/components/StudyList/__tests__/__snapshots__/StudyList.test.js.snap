// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    class="ui basic segment ui container stackable grid"
  >
    <div
      class="eleven wide computer eight wide tablet column"
    >
      <h1
        class="ui header"
      >
        Your Studies
      </h1>
    </div>
    <div
      class="stretched right aligned middle aligned five wide computer eight wide tablet column"
    >
      <div
        class="ui buttons"
      >
        <button
          class="ui icon primary left labeled button"
          tabindex="0"
        >
          <i
            aria-hidden="true"
            class="grid layout icon"
          />
          Grid
        </button>
        <button
          class="ui icon left labeled button"
          tabindex="0"
        >
          <i
            aria-hidden="true"
            class="list icon"
          />
          List
        </button>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="column"
      >
        <div
          class="ui stackable four cards"
        >
          <a
            class="ui card"
            href="/study/SD_8WX8QQ06/documents"
          >
            <div
              class="content"
            >
              <div
                class="header"
              >
                SD_8WX8QQ06
              </div>
              <div
                class="description"
              >
                benchmark extensible e-business
              </div>
            </div>
            <div
              class="extra content"
            >
              <div
                class="ui horizontal list"
                role="list"
              >
                <div
                  class="item"
                  role="listitem"
                >
                  <i
                    aria-hidden="true"
                    class="file icon"
                  />
                  1
                   files
                </div>
                <div
                  class="item"
                  role="listitem"
                >
                  <div
                    class="ui orange mini circular empty label"
                  />
                   
                  1
                </div>
              </div>
            </div>
          </a>
          <a
            class="ui card"
            href="/study/SD_I1L92W57/documents"
          >
            <div
              class="content"
            >
              <div
                class="header"
              >
                SD_I1L92W57
              </div>
              <div
                class="description"
              >
                visualize dynamic users
              </div>
            </div>
            <div
              class="extra content"
            >
              <div
                class="ui horizontal list"
                role="list"
              >
                <div
                  class="item"
                  role="listitem"
                >
                  <i
                    aria-hidden="true"
                    class="file icon"
                  />
                  0
                   files
                </div>
              </div>
            </div>
          </a>
          <a
            class="ui card"
            href="/study/SD_SG5N41K8/documents"
          >
            <div
              class="content"
            >
              <div
                class="header"
              >
                SD_SG5N41K8
              </div>
              <div
                class="description"
              >
                strategize cross-media markets
              </div>
            </div>
            <div
              class="extra content"
            >
              <div
                class="ui horizontal list"
                role="list"
              >
                <div
                  class="item"
                  role="listitem"
                >
                  <i
                    aria-hidden="true"
                    class="file icon"
                  />
                  0
                   files
                </div>
              </div>
            </div>
          </a>
          <a
            class="ui card"
            href="/study/SD_0YYP5ZEN/documents"
          >
            <div
              class="content"
            >
              <div
                class="header"
              >
                SD_0YYP5ZEN
              </div>
              <div
                class="description"
              >
                drive distributed architectures
              </div>
            </div>
            <div
              class="extra content"
            >
              <div
                class="ui horizontal list"
                role="list"
              >
                <div
                  class="item"
                  role="listitem"
                >
                  <i
                    aria-hidden="true"
                    class="file icon"
                  />
                  0
                   files
                </div>
              </div>
            </div>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders loading state 1`] = `
<div>
  <div
    class="ui basic segment ui container"
  >
    <div
      class="ui basic segment ui container"
    >
      <div
        class="ui placeholder"
      >
        <div
          class="header"
        >
          <div
            class="line"
          />
        </div>
      </div>
    </div>
    <div
      class="ui stackable four cards"
    >
      <div
        class="ui card"
      >
        <div
          class="content"
        >
          <div
            class="ui placeholder"
          >
            <div
              class="header"
            >
              <div
                class="line"
              />
            </div>
            <div
              class="paragraph"
            >
              <div
                class="line long"
              />
            </div>
          </div>
        </div>
        <div
          class="extra content"
        >
          <div
            class="ui placeholder"
          >
            <div
              class="paragraph"
            >
              <div
                class="line short"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="ui card"
      >
        <div
          class="content"
        >
          <div
            class="ui placeholder"
          >
            <div
              class="header"
            >
              <div
                class="line"
              />
            </div>
            <div
              class="paragraph"
            >
              <div
                class="line long"
              />
            </div>
          </div>
        </div>
        <div
          class="extra content"
        >
          <div
            class="ui placeholder"
          >
            <div
              class="paragraph"
            >
              <div
                class="line short"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="ui card"
      >
        <div
          class="content"
        >
          <div
            class="ui placeholder"
          >
            <div
              class="header"
            >
              <div
                class="line"
              />
            </div>
            <div
              class="paragraph"
            >
              <div
                class="line long"
              />
            </div>
          </div>
        </div>
        <div
          class="extra content"
        >
          <div
            class="ui placeholder"
          >
            <div
              class="paragraph"
            >
              <div
                class="line short"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="ui card"
      >
        <div
          class="content"
        >
          <div
            class="ui placeholder"
          >
            <div
              class="header"
            >
              <div
                class="line"
              />
            </div>
            <div
              class="paragraph"
            >
              <div
                class="line long"
              />
            </div>
          </div>
        </div>
        <div
          class="extra content"
        >
          <div
            class="ui placeholder"
          >
            <div
              class="paragraph"
            >
              <div
                class="line short"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
